### PR TITLE
Security/stripe

### PR DESCRIPTION
**Pull request by antoineeudes**

### CODE REVIEW

File: functions/payment.js [hunk 1]
```diff
@@ -31,33 +44,33 @@ exports.createCustomerWithPaymentMethod = functions.https.onCall(
...
+      throw new functions.https.HttpsError("unavailable", err.message);
+    }
+
+    await admin
+      .firestore()
+      .collection(CUSTOMER_TABLE)
+      .doc(context.auth.token.user_id)
+      .set({
+        userId: context.auth.token.user_id,
+        custormerId: customer.id,
```
Comment (maxb94) [inline functions/payment.js:71]: customerId

File: functions/payment.js [hunk 2]
```diff
@@ -86,8 +99,10 @@ exports.createPaymentIntent = functions.https.onCall(async (data, context) => {
     );
   }
 
+  const eventPrice = getEventPrice(data.venueId, data.eventId);
```
Comment (mathieud) [inline functions/payment.js:102]: What if this doesn't exist?

File: src/utils/hasUserBoughtTicket.ts [hunk 3]
```diff
@@ -6,4 +6,6 @@ export const hasUserBoughtTicketForEvent = (
   eventId: string | ParsedQs | string[] | ParsedQs[]
 ) =>
   userPurchaseHistory &&
-  !!userPurchaseHistory.find((purchase) => purchase.eventId === eventId);
+  !!userPurchaseHistory.find(
```
Comment (mathieud) [inline src/utils/hasUserBoughtTicket.ts:9]: Side note, but this makes double payment a bit more likely (while allowing someone who left the payment flow to buy a new one)

File: src/components/molecules/EventPaymentButton/EventPaymentButton.tsx [hunk 4]
```diff
@@ -15,13 +15,15 @@ interface PropsType {
...
   selectEvent: () => void;
+  isPaidSuccessfully: boolean;
 }
 
 const EventPaymentButton: React.FunctionComponent<PropsType> = ({
   event,
   venueId,
   setIsPaymentModalOpen,
   selectEvent,
+  isPaidSuccessfully,
```
Comment (mathieud) [inline src/components/molecules/EventPaymentButton/EventPaymentButton.tsx:26]: Not great naming. `paymentConfirmationPending`?

File: src/components/organisms/PaymentModal/PaymentModal.tsx [hunk 5]
```diff
@@ -55,23 +58,21 @@ const PaymentModal: React.FunctionComponent<PropsType> = ({
...
-        setIsPaymentSuccess={setIsPaymentSuccess}
+        setIsPaymentSuccess={() => setEventPaidSuccessfully(selectedEvent.id)}
         setIsPaymentProceeding={setIsPaymentProceeding}
         setIsCardBeingSaved={setIsCardBeingSaved}
         isPaymentProceeding={isPaymentProceeding}
         isCardBeingSaved={isCardBeingSaved}
         event={selectedEvent}
       />
     );
-  } else {
```
Comment (mathieud) [inline src/components/organisms/PaymentModal/PaymentModal.tsx:73]: Why remove this? It shouldn't be possible but who knows.

Comment (mathieud) [reply src/components/organisms/PaymentModal/PaymentModal.tsx:73]: Or always display PaymentConfirmation if `hasUserBoughtTicket`. Here it seems we have a case where the users would not see anything

**Comment by mathieud** [comment]
@antoineeudes We should also have a backend check to avoir double payment (if you already have one completed for this event do not create a new intent)

**Pull request merged**